core/app/services/spree/checkout/get_shipping_rates.rb

```diff
@@ -45,6 +45,7 @@ def return_shipments(order:)
         success(order.shipments.includes([shipping_rates: :shipping_method]))
       end
 
+      # this should be removed/moved/changed
       def move_order_to_delivery_state(order:)
         Spree::Dependencies.checkout_next_service.constantize.call(order: order) until order.state == 'delivery'
```

Comment:
It's crucial to always push order to `delivery` state in requesting for `shipping_rates` ?